packages/vats/src/core/startWalletFactory.js

```diff
@@ -0,0 +1,135 @@
+// @ts-check
+import { E, Far } from '@endo/far';
+import { deeplyFulfilled } from '@endo/marshal';
+
+import * as BRIDGE_ID from '../bridge-ids.js';
+import { makeStorageNodeChild } from '../lib-chainStorage.js';
+import { makeMyAddressNameAdmin, PowerFlags } from './basic-behaviors.js';
+
+const { details: X } = assert;
+
+/**
+ * @param {ERef<ZoeService>} zoe
+ * @param {Installation<import('@agoric/smart-wallet/src/walletFactory').start>} inst
+ * @typedef {Awaited<ReturnType<typeof startFactoryInstance>>} WalletFactoryStartResult
+ */
+// eslint-disable-next-line no-unused-vars
+const startFactoryInstance = (zoe, inst) => E(zoe).startInstance(inst);
+
+/**
+ * Register for PLEASE_PROVISION bridge messages and handle
+ * them by providing a smart wallet from the wallet factory.
+ *
+ * @param {BootstrapPowers} param0
+ */
+export const startWalletFactory = async ({
+  consume: {
+    agoricNames,
+    bankManager,
+    board,
+    bridgeManager: bridgeManagerP,
+    chainStorage,
+    namesByAddress,
+    namesByAddressAdmin: namesByAddressAdminP,
+    zoe,
+  },
+  produce: { client, walletFactoryStartResult },
+  installation: {
+    consume: { walletFactory },
+  },
+}) => {
+  const STORAGE_PATH = 'wallet';
+
+  const [storageNode, bridgeManager, namesByAddressAdmin] = await Promise.all([
+    makeStorageNodeChild(chainStorage, STORAGE_PATH),
+    bridgeManagerP,
+    namesByAddressAdminP,
+  ]);
+
+  const terms = await deeplyFulfilled(
+    harden({
+      agoricNames,
+      namesByAddress,
+      board,
+    }),
+  );
+  /** @type {WalletFactoryStartResult} */
```

Comment:
doesn't hurt but not necessary. `startInstance` now carries forward the type of the installation.
```suggestion
```